### DISCUSSION TITLE
infinite loop

### DISCUSSION THREAD

**Discussion by melassa** [astral-sh/ruff#4342] [Q&A]
I have a long python file (1800 lines) which send Ruff in infinite loop. 
After a trial error session, i reduced the problem to 2 configuration options: PLE and UP. If I exclude them in .toml Ruff works well with that python file.

Some suggestion?

**Comment by JonathanPlasse** [comment]
Can you share the file or a minimal reproducible example?

**Comment by melassa** [comment]
No I can't. But I accept suggestion about reducing code to "minimal reproducible example". For example, can I activate a log in Ruff to see where it stops? (I'm a new Ruff user, started to use it just yesterday). I can add that pylint works well with the same python file (PLE == pylint).

**Reply by charliermarsh** [reply]
Could you include the Python file that you’re running on?

**Reply by melassa** [reply]
Yes I can import it in another python script without problem, if is this that you asked.

**Reply by charliermarsh** [reply]
Are you able to include the contents of the file here in this issue, or a link to it? It’s hard for us to debug or fix the issue if we don’t have access to example code that triggers it.

**Reply by melassa** [reply]
Sorry, ss I said is a 1800 lines files for internal use, I can't share it.
I can add that i do not see ANY output from Ruff when I start it with the file.  I must stop Ruff with control-c from command prompt under window.
I repeat: there is a debug mode or a log mode to be activated in Ruff?

**Reply by melassa** [reply]
I can try to delete some code from the script until Ruff works, but it is not a fast work. :-)

**Reply by MichaReiser** [reply]
You can try to do a binary search to speed this up.

* Comment out one half the lines
* Test if ruff runs fine
	* No: Uncomment the lines. Try commenting the other half of the lines
	* Yes: Leave the lines commented, Repeat process with the remaining lines

**Reply by melassa** [reply]
I'll try. ASAP

**Reply by melassa** [reply]
Found the problem for PLE: is a really complex f-string. Now i try to reduce it to minimal example.

**Comment by melassa** [comment]
Well, this is a minimal example: paste this f-string as is in an empty file and ruff will not return. The configuration is:
```
[tool.ruff]
select = ["PLE"]

``````
f"""

    {x.join("'%s'"
    for f in v())}

    {a}

    {x.join("'%s' _%s" % (s,s) for s in t)}
"""

```

**Reply by MichaReiser** [reply]
Awesome. Thank you so much for taking the time to create a minimal reproducible example.

**Reply by melassa** [reply]
Thank to all developers for Ruff!

**Reply by konstin** [reply]
i unfortunately can't reproduce this:

code.py:

```python
f"""

    {x.join("'%s'"
    for f in v())}

    {a}

    {x.join("'%s' _%s" % (s,s) for s in t)}
"""
```

```
$ ruff --version
ruff 0.0.265
$ ruff --no-cache --select PLE --select UP code.py 
code.py:8:13: UP031 [*] Use format specifiers instead of percent format
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

this is how the code looks like for me, could you please confirm that this is the right indentation and newlines?

![image](https://github.com/charliermarsh/ruff/assets/6826232/d8553fe4-df1d-4f92-9b5d-5e263833621c)

**Reply by melassa** [reply]
It seems the same to me. I tryed with your config options and Ruff does not return. The version is the same. My system is Windows.
Here a snippet: https://goonlinetools.com/snapshot/code/#6toc39a4cfcy6g731yqx78

**Comment by melassa** [comment]
A note: also spaces and empty lines are important. Do not change them .

**Comment by melassa** [comment]
Same result with this configuration:
```
[tool.ruff]
select = ["UP"]
```
